toolchain/sem_ir/inst.h

```diff
@@ -55,13 +50,10 @@ struct InstLikeTypeInfoBase {
 
 // A particular type of instruction is instruction-like.
 template <typename TypedInst>
-struct InstLikeTypeInfo<
-    TypedInst,
-    static_cast<bool>(
-        std::is_same_v<const InstKind::Definition<
-                           typename decltype(TypedInst::Kind)::TypedNodeId>,
-                       decltype(TypedInst::Kind)>)>
-    : InstLikeTypeInfoBase<TypedInst> {
+  requires std::is_same_v<const InstKind::Definition<
```

Comment:
Style guidance is to use named concepts when available instead of predicates, here and elsewhere:

```suggestion
  requires std::same_as<const InstKind::Definition<
```

Reply:
Done, in the code affected by this patch.

Do we want to use `std::same_as` instead of `std::is_same_v` even in places unrelated to constraints, such as in `if constexpr` checks?

Reply:
I don't feel strongly at all, but I would probably just switch to `std::same_as` where it applies for consistency. It seems useful in constraints, and not worth the confusion of having a mixture.

That said, totally open to other approaches here too.

Reply:
Also, I'd switch to include `<concepts>` -- ClangD doesn't seem to always get this right as yet.

Reply:
Done.